koku/api/metrics/views.py

```diff
@@ -45,14 +51,29 @@ def get_queryset(self):
 
         Restricts the returned data to provider_uuid if supplied as a query parameter.
         """
-        queryset = CostModelMetricsMap.objects.all()
+        try:
+            with open(COST_MODEL_METRICS_FILE_NAME) as json_file:
+                queryset = json.load(json_file)
+        except (json.JSONDecodeError, OSError):
+            raise CostModelMetricMapJSONException("Internal Error loading JSON")
+
         source_type = self.request.query_params.get("source_type")
         if source_type:
-            queryset = queryset.filter(source_type=source_type)
+            # Query on source type
+            queryset = list(filter(lambda x: x.get("source_type") == source_type, queryset))
 
         return queryset
 
     @vary_on_headers(RH_IDENTITY_HEADER)
     def list(self, request, *args, **kwargs):
         """Obtain the list of CostModelMetrics for the tenant."""
         return super().list(request=request, args=args, kwargs=kwargs)
+
+
+class CostModelMetricMapJSONException(APIException):
```

Comment:
What benefit does this class have over just returning a status Response with 500 and an error?

Reply:
Andrew said: "Take a look at https://github.com/project-koku/koku/blob/master/koku/cost_models/view.py#L87-L93 so that’s the form of error message we want to return when something goes wrong.
koku/cost_models/view.py:87-93
```
class CostModelProviderQueryException(APIException):
    """Rate query custom internal error exception."""

    def __init__(self, message):
        """Initialize with status code 500."""
"
```

Reply:
👍